adapter/slack.py

```diff
@@ -112,3 +113,22 @@ async def list_bots(self, request: Request):
             return Response(status_code=200)
         except SlackApiError as e:
             raise HTTPException(status_code=400, detail=f"Slack API Error : {e}")
+    
+    def event_message(self, event, say):
+        if 'thread_ts' in event:
+            self.event_message_replied(event, say)
+    
+    def event_message_replied(self, event, say):
+        parent_message = self.app.client.conversations_history(

```

Comment:
do you mind explaining what this line of code trying to do ?

Reply:
It's for retrieving the parent thread message. Because we want to make sure that the we only reply to the thread that started with the `/ask` command, we have to retrieve the parent message first.

Here's a link to explain the method: https://api.slack.com/messaging/retrieving#individual_messages